packages/amplify-provider-awscloudformation/src/aws-utils/aws-cfn.js

```diff
@@ -153,7 +155,9 @@ class CloudFormation {
   }
 
   readStackEvents(stackName) {
-    this.pollForEvents = setInterval(() => this.addToPollQueue(stackName, 3), CFN_POLL_TIME);
+    const waitTimeInMs = Math.min((2 ^ this.readStackEventsCalls) * CFN_POLL_TIME, CFN_POLL_TIME_MAX);
```

Comment:
can this overflow if we hit enough readStackEventsCalls count ?
consider short circuiting this when we hit max.